app/scripts/lib/metrics.js

```diff
@@ -303,6 +304,18 @@ define(function (require, exports, module) {
     },
 
     /**
+     * Log an event only if it never happened before during this page load.
+     *
+     * @param {String} eventName
+     */
+    logEventOnce: function (eventName) {
+      if (_.indexOf(this._eventMemory, eventName) === -1) {
```

Comment:
See the previous comment about searching `this.events` directly instead of storing a local list.